jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -0,0 +1,277 @@
+   * DOM listeners
+   */
+
+  onFileDrop(files) {
+    if (files.length >= 1) {
+      this.setState({isUploadFilePending: true});
+
+      files.forEach(async (file) => {
+        var base64File = await this.toBase64(file);
+	    var formMediaJSON = {
+	      description: 'default',
+          file_type: 'form_media',
+          metadata: JSON.stringify({filename: file.name}),
+	      base64Encoded: base64File
+	    };
+        this.uploadMedia(formMediaJSON);
+      });
+    }
+  }
+
+  onInputURLChange(inputURL) {
+    this.setState({inputURL: inputURL});
+  }
+
+  onSubmitURL() {
+    var url = this.state.inputURL;
+
+    if (url === '') {
+      alertify.warning(t('URL is empty!'));
+    } else {
+      this.setState({
+        isUploadURLPending: true,
+        inputURL: ''
+      });
+
+      var formMediaJSON = {
+        description: 'default',
+        file_type: 'form_media',
+        metadata: JSON.stringify({redirect_url: url}),
+      };
+      this.uploadMedia(formMediaJSON);
+    }
+  }
+
+  onDeleteMedia(url) {
+    actions.media.deleteMedia(this.props.asset.uid, url);
+  }
+
+  uploadedAssetsIsEmpty() {
+    return (
+      this.state.uploadedAssets === null ||
+      this.state.uploadedAssets === undefined
+    );
+  }
+
+  /*
+   * rendering
+   */
+
+  renderButton() {
+    const buttonClassNames = ['formBuilder-header__button', 'formBuilder-header__button--save'];
+    if (this.state.isUploadURLPending) {
+      buttonClassNames.push('formBuilder-header__button--savepending');
+    }
+    return (
+      <button className={buttonClassNames.join(' ')}>
+        <i />
+        {t('ADD')}
+      </button>
+    );
+  }
+
+  renderFileName(item){
+    // Check if current item is uploaded vis URL. `redirect_url` is the indicator
+    var fileName = item.metadata.filename;
+    if (item.metadata.redirect_url) {
+      // Shorten URL to exactly 50 chars
+      var url = item.metadata.redirect_url;
+      var urlBack = url.slice(url.length - 22);
+      var urlFront = url.replace('https://', '').replace('http://', '').substr(0, 25);
+      fileName = urlFront + '...' + urlBack;
+    }
+
+    return (
+      <a href={item.content} target='_blank'>{fileName}</a>
+    );
+  }
+
+  renderIcon(item) {
+    const iconClassNames = ['form-media__file-type', 'fa'];
+    // Check if current item is uploaded vis URL. `redirect_url` is the indicator
+    if (item.metadata.redirect_url) {
+      iconClassNames.push('fa-link');
+    } else {
+      iconClassNames.push('fa-file');
+    }
+
+    return (
+      <i className={iconClassNames.join(' ')}/>
+    );
+  }
+
+  renderLoading(message = t('loading…')) {
+    return (
+      <bem.Loading>
+        <bem.Loading__inner>
+          <i />
+          {message}
+        </bem.Loading__inner>
+      </bem.Loading>
+    );
+  }
+
+  render() {
+    return (
+      <bem.FormModal__form className='project-settings project-settings--upload-file media-settings--upload-file' onSubmit={this.onSubmitURL}>
```

Comment:
Also I see that other sections use `bem.FormView`, not `bem.FormModal__form`, so to sum up, this should become:

```
<bem.FormView m='form-media' className='form-media`>
```

And then any changes needed over `form-view` you will apply in `.form-view--form-media` (probably only white background?), and everything else related to `form-media` should be inside `.form-media`. It would be a good idea to actually split all those styles to a new file `_kobo.form-media.scss`

Reply:
Not having `project-settings` here, you will need to move the `.project-settings .dropzone` styles to be shared with `.form-media`